File: py-crawler/classes/databases/twitter_accounts.py
from sqlalchemy import Column
from sqlalchemy import String, Date, Integer, Text, ForeignKey

from .base import Base

# twitter_accounts table.
# Author: Antonio Scardace

class TwitterAccounts(Base):

    __tablename__ = 'twitter_accounts'

    handle = Column('handle', String(255), nullable=False, primary_key=True)
    id = Column('id', String(64), nullable=False)
    full_name = Column('full_name', String(64), nullable=False)
    profile_image_url = Column('profile_image_url', String(255), nullable=False)
    verified = Column('verified', Integer, nullable=False)
    created_on = Column('created_on', String(64), nullable=False)
    followers_count = Column('followers_count', Integer, nullable=False)
    following_count = Column('following_count', Integer, nullable=False)
    tot_tweets_count = Column('tot_tweets_count', Integer, nullable=False)
    description = Column('description', Text, nullable=False)
    name = Column('name', String(32), ForeignKey('political_groups.name', ondelete='CASCADE'), nullable=False)
    last_update = Column('last_update', Date, nullable=False)

    def __init__(self, handle: str):
        self.handle = handle

File: node-ui/classes/Database/TwitterAccounts.js
const { DataTypes } = require('sequelize');
const sequelize = require('./base');
const PoliticalGroups = require('./PoliticalGroups');

const TwitterAccounts = sequelize.define('twitter_accounts', {
    handle: {
        type: DataTypes.STRING(255),
        allowNull: false,
        primaryKey: true
    },
    id: {
        type: DataTypes.STRING(64),
        allowNull: false
    },
    full_name: {
        type: DataTypes.STRING(64),
        allowNull: false
    },
    profile_image_url: {
        type: DataTypes.STRING(255),
        allowNull: false
    },
    verified: {
        type: DataTypes.BOOLEAN,
        allowNull: false
    },
    created_on: {
        type: DataTypes.STRING(64),
        allowNull: false
    },
    followers_count: {
        type: DataTypes.INTEGER(11),
        allowNull: false
    },
    following_count: {
        type: DataTypes.INTEGER(11),
        allowNull: false
    },
    tot_tweets_count: {
        type: DataTypes.INTEGER(11),
        allowNull: false
    },
    description: {
        type: DataTypes.TEXT,
        allowNull: false
    },
    last_update: {
        type: DataTypes.DATEONLY,
        allowNull: false
    },
    name: {
        type: DataTypes.STRING(32),
        allowNull: false,
        references: {
            model: PoliticalGroups,
            key: 'name',
        },
        onDelete: 'CASCADE'
    }
});

TwitterAccounts.belongsTo(PoliticalGroups, {foreignKey: 'name', onDelete: 'CASCADE'});

module.exports = TwitterAccounts